lib/srv/discovery/access_graph.go

```diff
@@ -320,3 +301,35 @@ func (s *Server) initAccessGraphWatchers(ctx context.Context, cfg *Config) error
 	}
 	return nil
 }
+
+// accessGraphFetchersFromMatchers converts Matchers into a set of AWS Sync Fetchers.
+func (s *Server) accessGraphFetchersFromMatchers(ctx context.Context, matchers Matchers) ([]aws_sync.AWSSync, error) {
+	var fetchers []aws_sync.AWSSync
+	var errs []error
+	if matchers.AccessGraph != nil && len(matchers.AccessGraph.AWS) > 0 {
```

Comment:
:nail_care: 
```suggestion
	if matchers.AccessGraph != nil {
```
The range iteration is safe against an empty slice.